lnbits/static/js/event-reactions.js

```diff
@@ -1,16 +1,30 @@
-function eventReactionWebocket(event_id, event_function) {
+function eventReactionWebocket(event_id) {
   localUrl = ''
+  reaction = localStorage.getItem('lnbits.reactions')
+  if (reaction == null || reaction == 'None' || reaction == undefined) {
+    return
+  }
   if (location.protocol !== 'http:') {
     localUrl = 'wss://' + location.host + '/api/v1/ws/' + event_id
   } else {
     localUrl = 'ws://' + location.host + '/api/v1/ws/' + event_id
   }
   connection = new WebSocket(localUrl)
   connection.onmessage = function (e) {
-    event_function()
+    let parsedData = JSON.parse(e.data)
+    if (parsedData.payment.amount < 0) {
+      return
+    } else {
+      try {
+        reaction = localStorage.getItem('lnbits.reactions')
+        eval(reaction.split('|')[1] + '()')
```

Comment:
`eval` is dangerous, we should not use it:
https://www.digitalocean.com/community/tutorials/js-eval